### ISSUE TITLE
Something strange in save_file_to_file code

### ISSUE THREAD

**Issue by fviard** [gphoto/gphoto2#41]
I was randomly looking at gphoto2 code when I saw something very weird that I guess is testing code that was forgotten.

Look at the following line of main.c : save_file_to_file:
https://github.com/gphoto/gphoto2/blob/ae995fe0966dbe870b202b4e68eb1770c512fa6a/gphoto2/main.c#L543

It looks like that for downloading files content, you are randomly choosing between libgphoto2 "fd" or "handler" method. If I suppose that both methods work correctly in all situations, that looks a little weird for me and could be cause of random non understandable issues?

This was introduced by the following change:
https://github.com/gphoto/gphoto2/commit/ae995fe0966dbe870b202b4e68eb1770c512fa6a#diff-58ad5a323f09c958fa9666f43f1cf7f7R543

**Comment by msmeissn** [comment]
yes, the idea was to increase coverage of testing the two methods. 

yes, probably should consider changing this and doing the method switching differently
